### PR TITLE
feat(uat): implement T17 and T18 scenario outlines

### PR DESCRIPTION
**Pull request by bgklika**
**Issue #, if available:**
https://klika-tech.atlassian.net/browse/GGMQ-199
Implement GGMQ scenario based on GGAD-1-T17, GGAD-1-T18

**Description of changes:**
- Add component aws.greengrass.clientdevices.mqtt.Bridge v2.2.6 to local store
- Add GGMQ-1-T17
- Add GGMQ-1-T18
- Tune existing recipe files

**Why is this change necessary:**
Make replacement of GGAD-1-T17 and GGAD-1-T18

**How was this change tested:**
Run scenario on CodeBuild

**Test results:**
T18 (Now clients reduced to only sdk-java)
```
[INFO ] 2023-06-28 19:12:54.568 [main] StepTrackingReporting - Passed: 'GGMQ-1-T18-v3-sdk-java: As a customer, I can configure IoT Core messages to be forwarded to local MQTT topic, with Greengrass in disconnected state'
[INFO ] 2023-06-28 19:12:54.568 [main] StepTrackingReporting - Passed: 'GGMQ-1-T18-v3-mosquitto-c: As a customer, I can configure IoT Core messages to be forwarded to local MQTT topic, with Greengrass in disconnected state'
[INFO ] 2023-06-28 19:12:54.569 [main] StepTrackingReporting - Passed: 'GGMQ-1-T18-v3-paho-java: As a customer, I can configure IoT Core messages to be forwarded to local MQTT topic, with Greengrass in disconnected state'
[INFO ] 2023-06-28 19:12:54.569 [main] StepTrackingReporting - Passed: 'GGMQ-1-T18-v5-sdk-java: As a customer, I can configure IoT Core messages to be forwarded to local MQTT topic, with Greengrass in disconnected state'
[ERROR] 2023-06-28 19:12:54.569 [main] StepTrackingReporting - Failed: 'GGMQ-1-T18-v5-mosquitto-c: As a customer, I can configure IoT Core messages to be forwarded to local MQTT topic, with Greengrass in disconnected state': Failed at 'the Greengrass deployment is COMPLETED on the device after 5 minutes'
[INFO ] 2023-06-28 19:12:54.569 [main] StepTrackingReporting - Passed: 'GGMQ-1-T18-v5-paho-java: As a customer, I can configure IoT Core messages to be forwarded to local MQTT topic, with Greengrass in disconnected state'
```
T17:
```
[INFO ] 2023-06-30 03:58:43.667 [main] StepTrackingReporting - Passed: 'GGMQ-1-T17-v3-sdk-java: As a customer, I can configure IoT Core messages to be forwarded to local MQTT topic, with Greengrass in offline st
ate'
[INFO ] 2023-06-30 03:58:43.668 [main] StepTrackingReporting - Passed: 'GGMQ-1-T17-v5-sdk-java: As a customer, I can configure IoT Core messages to be forwarded to local MQTT topic, with Greengrass in offline st
ate'
```

**Any additional information or context required to review the change:**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: uat/testing-features/src/main/java/com/aws/greengrass/platforms/PlatformResolver.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        if (Files.exists(Paths.get("/usr/bin/apt-get"))) {
+            ranks.put("debian", 11);
+        }
+        if (IS_WINDOWS) {
+            ranks.put("windows", 5);
+        }
+        if (Files.exists(Paths.get("/usr/bin/yum"))) {
+            ranks.put("fedora", 11);
+        }
+        if (!IS_WINDOWS) {
```
Comment (auarbekov) [inline uat/testing-features/src/main/java/com/aws/greengrass/platforms/PlatformResolver.java]: I guess we need to remove unnecessary checks
```
if (IS_WINDOWS) {
  ranks.put("windows", 5);
} else {
}
```

File: uat/testing-features/src/main/java/com/aws/greengrass/platforms/PlatformResolver.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                }
+                if (sysver.contains("cygwin")) {
+                    ranks.put("cygwin", 22);
+                }
+                if (sysver.contains("freebsd")) {
+                    ranks.put("freebsd", 22);
+                }
+                if (sysver.contains("solaris") || sysver.contains("sunos")) {
+                    ranks.put("solaris", 22);
+                }
```
Comment (auarbekov) [inline uat/testing-features/src/main/java/com/aws/greengrass/platforms/PlatformResolver.java]: is there any chance that the variable 'sysver' has many os? if not we should use swith-case blocks.

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/NetworkUtilsSteps.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+    /**
+     * Convert offline/online string to boolean.
+     *
+     * @param value the string of offline/online value
+     * @throws RuntimeException on invalid value
+     */
+    @SuppressWarnings("PMD.UnnecessaryAnnotationValueElement")
+    @ParameterType(value = "offline|Offline|OFFLINE|online|Online|ONLINE")
+    public boolean connectivityValue(String value) {
```
Comment (auarbekov) [inline uat/testing-features/src/main/java/com/aws/greengrass/steps/NetworkUtilsSteps.java]: I think the name "status" would be more obvious than "value".

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/NetworkUtilsSteps.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+            case "OFFLINE":
+                return false;
+
+            case "online":
+            case "Online":
+            case "ONLINE":
+                return true;
+
+            default:
+                throw new RuntimeException("Invalid connectivity value " + value);
```
Comment (auarbekov) [inline uat/testing-features/src/main/java/com/aws/greengrass/steps/NetworkUtilsSteps.java]: I think **log.error** is necessary.

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/NetworkUtilsSteps.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    /**
+     * Disables or dnables device MQTT connectivity to IoT Core by blocking traffic on ports 8883 and 443.
+     *
+     * @param connectivity the value of connectivity to set
+     * @throws IOException on IO errors
+     * @throws InterruptedException when thread has been interrupted
+     */
+    @When("I set device mqtt connectivity to {connectivityValue}")
+    public void setDeviceMqtt(final boolean connectivity) throws IOException, InterruptedException {
+        boolean old = mqttConnectivity.getAndSet(connectivity);
```
Comment (auarbekov) [inline uat/testing-features/src/main/java/com/aws/greengrass/steps/NetworkUtilsSteps.java]: minor: Using prefix 'is' is the best practice, like **isOld**

File: uat/testing-features/src/main/java/com/aws/greengrass/platforms/PlatformResolver.java (hunk 1 above)
Comment (bgklika) [reply uat/testing-features/src/main/java/com/aws/greengrass/platforms/PlatformResolver.java]: extra checks are removed in f820a556c989ecf5ecb2a117bc62f3124ae7d774

File: uat/testing-features/src/main/java/com/aws/greengrass/platforms/PlatformResolver.java (hunk 2 above)
Comment (bgklika) [reply uat/testing-features/src/main/java/com/aws/greengrass/platforms/PlatformResolver.java]: extra checks are removed in f820a556c989ecf5ecb2a117bc62f3124ae7d774

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/NetworkUtilsSteps.java (hunk 3 above)
Comment (bgklika) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/NetworkUtilsSteps.java]: fixed in f820a556c989ecf5ecb2a117bc62f3124ae7d774

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/NetworkUtilsSteps.java (hunk 4 above)
Comment (bgklika) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/NetworkUtilsSteps.java]: added in f820a556c989ecf5ecb2a117bc62f3124ae7d774

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/NetworkUtilsSteps.java (hunk 5 above)
Comment (bgklika) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/NetworkUtilsSteps.java]: renamed to oldConnectivity in 820a556c989ecf5ecb2a117bc62f3124ae7d774

**Review by auarbekov — DISMISSED** [review]

**Review by eschastlivtsev — DISMISSED** [review]

File: uat/testing-features/src/main/java/com/aws/greengrass/platforms/common/Utils.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+package com.aws.greengrass.platforms.common;
+
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+
+/**
+ * Miscellaneous standalone functions for general usage.
+ */
+public final class Utils {
```
Comment (auarbekov) [inline uat/testing-features/src/main/java/com/aws/greengrass/platforms/common/Utils.java]: could we change the name to be GeneralUtils or something?

Comment (bgklika) [reply uat/testing-features/src/main/java/com/aws/greengrass/platforms/common/Utils.java]: Will removed soon

Comment (bgklika) [reply uat/testing-features/src/main/java/com/aws/greengrass/platforms/common/Utils.java]: removed in 60fcb0773b001cdb02d00947e33e5c6adb366253

**Review by eschastlivtsev — DISMISSED** [review]

**Review by auarbekov — DISMISSED** [review]

**Review by auarbekov — DISMISSED** [review]

**Review by eschastlivtsev — DISMISSED** [review]

**Review by auarbekov — DISMISSED** [review]

**Review by eschastlivtsev — DISMISSED** [review]

**Review by mzyl13 — DISMISSED** [review]
Fixed

**Pull request closed without merging**